qa-test/src/bin/embassy_wifi_i2s.rs

```diff
@@ -0,0 +1,287 @@
+//% FEATURES: esp-radio esp-radio/wifi esp-radio/smoltcp esp-radio/unstable esp-hal/unstable
+//% CHIPS: esp32 esp32s2 esp32s3
+
+#![no_std]
+#![no_main]
+
+extern crate alloc;
+
+use embassy_executor::Spawner;
+use embassy_net::{Runner, StackResources};
+use embassy_sync::{blocking_mutex::raw::NoopRawMutex, signal::Signal};
+use embassy_time::{Duration, Instant, Timer};
+use esp_alloc;
+use esp_backtrace as _;
+#[cfg(any(feature = "esp32", feature = "esp32s3"))]
+use esp_hal::interrupt::software::SoftwareInterruptControl;
+use esp_hal::{
+    dma_buffers,
+    i2s::master::{Channels, Config as I2sConfig, DataFormat, I2s},
+    rng::Rng,
+    time::Rate,
+    timer::timg::TimerGroup,
+};
+use esp_println::println;
+use esp_radio::{
+    Controller,
+    wifi::{
+        ClientConfig,
+        Config as WifiConfig,
+        WifiController,
+        WifiDevice,
+        WifiEvent,
+        WifiStaState,
+        sta_state,
+    },
+};
+use static_cell::StaticCell;
+
+esp_bootloader_esp_idf::esp_app_desc!();
+
+const SSID: &str = env!("SSID");
+const PASSWORD: &str = env!("PASSWORD");
+
+const SAMPLE_RATE: u32 = 16_000;
+const I2S_BUFFER_SIZE: usize = 4092 * 8; // DMA ring size
```

Comment:
Note to self: this example allocated much more static RAM than usual, maybe the crashes I see are stack overflows.